src/py-opentimelineio/opentimelineio/adapters/filebundle.py

```diff
@@ -0,0 +1,123 @@
+"""OTIOZ adapter - bundles otio files linked to local media
+
+Takes as input an OTIO file that has media references which are all relative
+local paths (ie file:///foo.mov) and bundles those files and the otio file into
+a single zip file with the suffix .otioz.  Can error out if files aren't
+locally referenced or provide missing references
+"""
+
+import os
+import copy
+import zipfile
+
+from .. import (
+    exceptions,
+    schema,
+)
+
+from . import (
+    otio_json
+)
+
+
+class NotAFileOnDisk(exceptions.OTIOError):
+    pass
+
+
+def read_from_file(filepath):
+    if not zipfile.is_zipfile(filepath):
+        raise exceptions.OTIOError("Not a zipfile: {}".format(filepath))
+
+    internal_playlist_file = "{0}/{0}.otio".format(
+        os.path.splitext(os.path.basename(filepath))[0]
+    )
+    with zipfile.ZipFile(filepath, 'r') as zi:
+        result = otio_json.read_from_string(zi.read(internal_playlist_file))
+
+    return result
+
+
+class MediaReferencePolicy:
+    ErrorIfNotFile = 0
+    MissingIfNotFile = 1
+
+
+def write_to_file(
+    input_otio,
+    filepath,
+    unreachable_media_policy=MediaReferencePolicy.ErrorIfNotFile,
+    dryrun=False
+):
+    referenced_files = set()
+
+    for cl in input_otio.each_clip():
+        try:
+            target_file = cl.media_reference.target_url
+        except AttributeError:
+            continue
+
+        if target_file in referenced_files:
+            continue
+
+        if not os.path.exists(target_file) or not os.path.isfile(target_file):
```

Comment:
`target_file` is `media_reference.target_url` which is suposdely an 'url' not a path, shouldn't you check that the protocol is `file` (make the assumption it is file if for some reason the url is a path) and then work with the URL path?